### PR TITLE
Added HDD serial numbers.

### PR DESCRIPTION
**Pull request by ghost**
Added HDD serial numbers to facts.py.

### CODE REVIEW

**Comment by Jmainguy** [comment]
I can confirm this does work and find the serial. It requires sg_inq to be installed on the target to gather HDD specific facts, if it is not installed it just doesn't provide them, does not spit out any ugly errors.

Good job @jlstanton  keep it up.

**Comment by jimi-c** [comment]
@jlstanton I would go ahead and remove the for loop, as it doesn't do anything in this case. Beyond that the patch looks fine.

**Comment by ghost** [comment]
@jimi-c, good point. I removed the for loop.

**Comment by jfchevrette** [comment]
Hmm I could actually have used that feature yesterday when I had to identify disks on about ~100 servers.
LGTM

**Comment by ghost** [comment]
.

**Comment by amenonsen** [comment]
Looks fine to me.

File: lib/ansible/module_utils/facts.py [hunk 1]
```diff
@@ -765,6 +765,17 @@ def get_device_facts(self):
                     continue
             d = {}
             diskname = os.path.basename(sysdir)
+
+            sg_inq = module.get_bin_path('sg_inq') 
+            device = "/dev/%s" % (block)
+            try:
+                rc, drivedata, err = module.run_command([sg_inq, device])
+            except:
+                return
```
Comment (bcoca) [inline lib/ansible/module_utils/facts.py:774]: this should not be a return as it breaks the whole for loop

File: lib/ansible/module_utils/facts.py [hunk 2]
```diff
@@ -765,6 +765,17 @@ def get_device_facts(self):
                     continue
             d = {}
             diskname = os.path.basename(sysdir)
+
+            sg_inq = module.get_bin_path('sg_inq') 
+            device = "/dev/%s" % (block)
+            try:
```
Comment (bcoca) [inline lib/ansible/module_utils/facts.py:771]: add if that skips it all if sq_inq is not found

**Comment by amenonsen** [comment]
I'd like to get this merged. @bcoca's comment about the return was correct, but no longer applies because the for loop has been removed, but his other comment about not trying to execute sg_inq if get_bin_path returns None still applies. @stantonnet sorry it's taken so long to get to this, but do you feel like making that change?

**Comment by amenonsen** [comment]
Anyone else using sg_inq interested in updating this patch?

**Comment by Jmainguy** [comment]
Sure, I will take a stab at updating this, will update here once done, if someone beats me to it, that is also very cool

**Comment by mattclay** [comment]
Closing since this was rebased and merged in #22852.

**Pull request closed without merging**